prowler/lib/persistence/sqlite.py

```diff
@@ -0,0 +1,267 @@
+import os
+import sqlite3
+import tempfile
+
+import dill
+
+DEFAULT_CACHE_SIZE = int(os.environ.get('PROWLER_CACHE_SIZE', 2000))
+
+
+class SQLiteDict:
+
+    def __init__(self, cache_size=2000):
+        self._tmp_path = tempfile.NamedTemporaryFile(prefix='prowler-dict-')
+        self.db_name = self._tmp_path.name
+        self.conn = sqlite3.connect(self.db_name)
```

Comment:
Same concerns as Pepe about creating a database and a connection every time we create a list or dict.